core/include/drivers/pinctrl.h

```diff
@@ -0,0 +1,94 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022, Microchip
+ */
+
+#ifndef __DRIVERS_PINCTRL_H
+#define __DRIVERS_PINCTRL_H
+
+#include <sys/queue.h>
+#include <tee_api_types.h>
+
+enum pin_config_mode {
+	PIN_MODE_BIAS_DISABLE,
+	PIN_MODE_BIAS_PULL_UP,
+	PIN_MODE_BIAS_PULL_DOWN,
+};
+
+/*
+ * struct pinctrl - Describe a pinctrl controller
+ *
+ * @node: Node offset of the controller in the device-tree
+ * @apply_state: callback which apply the content of a specific pinctrl conf
+ * @link: Link in list of pinctrl
+ */
+struct pinctrl {
+	int node;
+	TEE_Result (*apply_state)(struct pinctrl *pinctrl, const void *fdt,
+				  int node);
+	SLIST_ENTRY(pinctrl) link;
+};
+
+#ifdef CFG_DRIVERS_PINCTRL
+
+/**
+ * pinctrl_register - Register a pin controller
+ *
+ * @pinctrl: Structure representing the pin controller
+ */
+void pinctrl_register(struct pinctrl *pinctrl);
+
+/**
+ * pinctrl_apply_state - Apply a pinctrl state
+ *
+ * @fdt: Device-tree containing the pinctrl node
+ * @node: Node of the device for which to apply the pinctrl state
+ * @name: Name of the pinctrl state to apply or NULL for the "default" one
+ *
+ * Return a TEE_Result compliant value
+ */
+TEE_Result pinctrl_apply_state(const void *fdt, int node, const char *name);
+
+/**
+ * pinctrl_parse_dt_pin_mode_config - Parse a pinctrl configuration node
+ * Parse common device-tree pinctrl properties and return an array with values:
+ * - bias-disable
+ * - bias-pull-up
+ * - bias-pull-down
+ *
+ * @fdt: Device-tree which contains the pinctrl configuration node to parse
+ * @node: Pinctrl configuration node to parse
+ * @cfg: Returned configuration
+ * @count: Count of elements in the @cfg array.
+ *
+ * Return a TEE_Result compliant value
+ */
+TEE_Result pinctrl_parse_dt_pin_mode_config(const void *fdt, int node,
+					    uint32_t **cfg,
+					    unsigned int *count);
+#else
```

Comment:
You can recall here it's

#else /* CFG_DRIVERS_PINCTRL */

It helps reading :)

ditto for #endif